packages/core/test/financial-templates/funding-rate-store/FundingRateStore.js

```diff
@@ -1,74 +1,397 @@
+      const settlementTxn = await fundingRateStore.settleDispute(identifier, proposalTime, { from: disputer });
+
+      // Publish event was NOT emitted.
+      truffleAssert.eventNotEmitted(settlementTxn, "PublishedRate");
+
+      // Funding rate and publish time are linked to the proposal that expired in the middle of the dispute.
+      assert.equal((await fundingRateStore.getFundingRateForIdentifier(identifier)).rawValue.toString(), toWei("0.02"));
+      assert.equal(
+        (await fundingRateStore.getLastPublishTimeForIdentifier(identifier)).toString(),
+        midDisputeProposeTime.add(toBN(liveness)).toString()
+      );
+
+      // Disputed funding rate record is deleted.
+      const disputedProposal = await fundingRateStore.fundingRateDisputes(identifier, proposalTime);
+      isEmptyProposalStruct(disputedProposal.proposal);
+
+      // Disputer receives final fee rebate, proposer receives nothing.
+      const postBalanceDisputer = await collateralCurrency.balanceOf(disputer);
+      const postBalanceProposer = await collateralCurrency.balanceOf(proposer);
+      assert.equal(finalFeeAmount.toString(), toBN(postBalanceDisputer).sub(toBN(preBalanceDisputer)));
+      assert.equal("0", toBN(postBalanceProposer).sub(toBN(preBalanceProposer)));
+    });
+
+    it("Cannot dispute expired proposal", async function() {
+      await incrementTime(fundingRateStore, 1);
+      assert(await didContractThrow(fundingRateStore.dispute(identifier, { from: disputer })));
     });
   });
 
   describe("Expired Proposal", function() {
-    const identifier = toHex("expired-proposal");
+    const identifier = defaultTestIdentifier;
+    let proposeTime;
     beforeEach(async () => {
-      await fundingRateStore.propose(identifier, { rawValue: toWei("0.01") }, { from: account1 });
+      proposeTime = toBN(await fundingRateStore.getCurrentTime());
+      await fundingRateStore.propose(identifier, { rawValue: toWei("0.01") }, { from: proposer });
       await incrementTime(fundingRateStore, liveness);
     });
 
-    it("New rate is retrieved", async function() {
+    it("New rate and publish time are retrieved", async function() {
       assert.equal((await fundingRateStore.getFundingRateForIdentifier(identifier)).rawValue.toString(), toWei("0.01"));
+      assert.equal(
+        (await fundingRateStore.getLastPublishTimeForIdentifier(identifier)).toString(),
+        proposeTime.add(toBN(liveness)).toString()
+      );
     });
 
     it("New proposal allowed", async function() {
-      await fundingRateStore.propose(identifier, { rawValue: toWei("-0.01") }, { from: account1 });
+      await fundingRateStore.propose(identifier, { rawValue: toWei("-0.01") }, { from: proposer });
 
       // Double check that existing value still persists even after a fresh proposal.
       assert.equal((await fundingRateStore.getFundingRateForIdentifier(identifier)).rawValue.toString(), toWei("0.01"));
```

Comment:
these tests look good! Do you not want to also add a test that advances time past liveness and ensures that the `getFundingRateForIdentifier` returns the new funding rate?

Reply:
Technically the previous test tests for this already since it doesn't call `propose()` again. It only calls the view only method `getFundingRateForIdentifier` which returns the "updated" funding rate even though the `fundingRateRecords` struct hasn't been updated

Reply:
@chrismaree WDYT?

Reply:
fair enough. SGTM!